### PR TITLE
Tweak titles and descriptions of pages for SEO

### PR DESCRIPTION
**Pull request by YLivay**

### CODE REVIEW

File: components/form-vue/group/group.ts [hunk 1]
```diff
@@ -23,7 +23,12 @@ export default class AppFormGroup extends Vue {
 		const name = this.name;
 
 		if (!this.label) {
-			return titleCase(name);
+			return titleCase(name, {
```
Comment (hworld) [inline components/form-vue/group/group.ts:26]: Why is this here?

File: components/route/route-component.ts [hunk 2]
```diff
@@ -241,15 +243,18 @@ export class BaseRouteComponent extends Vue {
...
 		}
 
 		this.$watch('routeTitle', (title: string | null) => {
 			if (title) {
-				Meta.title = title;
+				Meta.setTitle(title, this.disableRouteTitleSuffix);
 			}
 		});
 
+		// TODO: should I watch disableRouteTitleSuffix?
```
Comment (hworld) [inline components/route/route-component.ts]: You should figure this TODO out before merging

File: components/route/route-component.ts [hunk 3]
```diff
@@ -357,6 +362,9 @@ export class BaseRouteComponent extends Vue {
 			const resolver = Resolver.startResolve(this.$options, to);
 			this.isRouteLoading = true;
 
+			// TODO: do we want to reset title suffix on route reload?
+			this.disableRouteTitleSuffix = false;
```
Comment (hworld) [inline components/route/route-component.ts:361]: do we?

File: utils/string.ts [hunk 4]
```diff
@@ -4,16 +4,66 @@ export function kebabCase(name: string) {
 	return name.replace(KebabCaseRegex, (letter, pos) => (pos ? '-' : '') + letter.toLowerCase());
 }
 
-export function titleCase(str: string) {
-	// Hyphen and underscore.
-	str = str.replace(/(\-|_)/g, ' ');
+export type TitleCaseOptions = Partial<{
```
Comment (hworld) [inline utils/string.ts:7]: You're doing this as part of the title changes?

File: utils/string.ts [hunk 5]
```diff
@@ -4,16 +4,66 @@ export function kebabCase(name: string) {
...
+	'On',
+	'Onto',
+	'Than',
+	'To',
+	'Upon',
+	'Via',
+	'With',
+];
+
+export function titleCase(str: string, options?: TitleCaseOptions) {
```
Comment (hworld) [inline utils/string.ts:45]: Should we get rid of multiple spaces in a row? `replace(/ +/, ' ')`

File: components/form-vue/group/group.ts (hunk 1 above)
Comment (YLivay) [reply components/form-vue/group/group.ts:26]: string utils title case was used specifically for group, which made no sense. I made the default behaviour of that function act more like a real title case with options to override it. these options will produce the same output group expected before that change.

File: components/route/route-component.ts (hunk 2 above)
Comment (YLivay) [reply components/route/route-component.ts]: I left it for the PR review. The issue is we watch routeTitle to figure out when we should set the Meta title. usually only within these funcs we'll modify the disableRouteTitleSuffix variable so the question is if that watch is ever needed. I can't currently think of a case where we'd modify disableRouteTitleSuffix outside of routeTitle.

File: components/route/route-component.ts (hunk 3 above)
Comment (YLivay) [reply components/route/route-component.ts:361]: If reloading a route changes the route suffix for some reason (should only be done from within routeTitle) then yeah, we'd need this. Thing is, same as the comment above - I don't know if such a use case will ever be needed. It's not harmful to keep it in there either. May as well..?

File: utils/string.ts (hunk 4 above)
Comment (YLivay) [reply utils/string.ts:7]: yup

File: utils/string.ts (hunk 5 above)
Comment (YLivay) [reply utils/string.ts:45]: I think so. This is meant to make a nice title, and multiple spaces aren't nice.
We shouldn't be running this function on user input like game titles anyways.

This comes in handy when we have intepolated strings in our translations that may be left empty, resulting in spaces on both sides for example.
